src/uct/ib/ud/base/ud_iface.c

```diff
@@ -157,22 +157,41 @@ static void uct_ud_iface_destroy_qp(uct_ud_iface_t *ud_iface)
     ops->destroy_qp(ud_iface);
 }
 
+static void uct_ud_iface_fill_qp_init_attr(uct_ib_iface_t *iface,
+                                           const uct_ud_iface_config_t *config,
+                                           uct_ib_qp_attr_t *qp_init_attr)
+{
+    uct_ib_md_t *ib_md   = ucs_derived_of(iface->super.md, uct_ib_md_t);
+    uct_ib_device_t *dev = &ib_md->dev;
+
+    memset(qp_init_attr, 0, sizeof(*qp_init_attr));
+
+    qp_init_attr->qp_type             = IBV_QPT_UD;
+    qp_init_attr->cap.max_send_wr     = config->super.tx.queue_len;
+    qp_init_attr->cap.max_recv_wr     = config->super.rx.queue_len;
+    qp_init_attr->cap.max_send_sge    = config->super.tx.min_sge + 1;
+    qp_init_attr->cap.max_recv_sge    = 1;
```

Comment:
this function seems like code duplication with uct_ib_iface_adjust_qp_attr

Reply:
Yeah, I made a mistake - should have only changed ib_iface.c ... I reverted all my changes to ud_iface.c in the following commit.